pages/library-card/review/index.tsx

```diff
@@ -16,37 +29,70 @@ import { findLibraryCode } from "../../../src/utils/FormValidationUtils";
 function ReviewPage() {
   const { handleSubmit } = useFormContext();
   const { state, dispatch } = useFormDataContext();
-  const { formValues, isLoading } = state;
+  const { formValues } = state;
   const router = useRouter();
 
-  const formKeys = Object.keys(formValues);
-  const getPatronAgencyType = (agencyTypeParam) => {
-    const { agencyType } = config;
-    return !isEmpty(agencyTypeParam) && agencyTypeParam.toLowerCase() === "nys"
-      ? agencyType.nys
-      : agencyType.default;
+  // Flags to set a section to editable or read-only.
+  const [editPersonalInfoFlag, setEditPersonalInfoFlag] = useState(false);
+  const [editAddressInfoFlag, setEditAddressInfoFlag] = useState(false);
+  const [editAccountInfoFlag, setEditAccountInfoFlag] = useState(false);
+
+  // Shareable button for each editable section.
+  const editButton = (editSectionFlag) => (
+    <Button
+      buttonType={ButtonTypes.Primary}
+      onClick={() => editSectionFlag(true)}
+    >
+      Edit
+    </Button>
+  );
+  const submitButton = (
+    <Button buttonType={ButtonTypes.Primary} onClick={() => {}} type="submit">
+      Submit
+    </Button>
+  );
+
+  /**
+   * editSectionInfo
+   * Each section has its own form and needs its own edit button, but the data
+   * is not being manipulated, just updated, so they can all use the same
+   * function to set the global data.
+   */
+  const editSectionInfo = (formData) => {
+    dispatch({
+      type: "SET_FORM_DATA",
+      value: {
+        ...formValues,
+        ...formData,
+      },
+    });
+    // Set all to false even though not all are on.
+    setEditPersonalInfoFlag(false);
+    setEditAccountInfoFlag(false);
+    setEditAddressInfoFlag(false);
   };
+
+  /**
+   * submitForm
+   * Update the form values again but this time submit to the API.
+   */
   const submitForm = (formData) => {
     // This is resetting any errors from previous submissions, if any.
     dispatch({ type: "SET_FORM_ERRORS", value: null });
-    // Render the loading component.
-    dispatch({ type: "SET_IS_LOADING", value: true });
 
-    formValues.homeLibraryCode = findLibraryCode(formValues.homeLibraryCode);
-    formValues.agencyType = getPatronAgencyType(formValues.location);
-
-    const dataToSubmit = { ...formData, ...formValues };
+    // Convert the home library name to its code value.
+    formData.homeLibraryCode = findLibraryCode(formData.homeLibraryCode);
 
     // Update the global state.
     dispatch({
       type: "SET_FORM_DATA",
-      value: dataToSubmit,
+      value: formData,
     });
 
     axios
       .post(
         "/api/create-patron",
-        dataToSubmit
+        formData
         // {
         //    headers: { "csrf-token": csrfToken },
```

Comment:
Was this supposed to stay commented out?

Reply:
Yes. It's been commented out since the start. The CSRF part is harder to implement in nextjs than it is in Express and I haven't found a good way to do it just yet.